src/server/game/AI/SmartScripts/SmartAI.cpp

```diff
@@ -188,33 +216,28 @@ void SmartAI::StopPath(uint32 DespawnTime, uint32 quest, bool fail)
 
     if (quest)
         mEscortQuestID = quest;
+
     SetDespawnTime(DespawnTime);
     //mDespawnTime = DespawnTime;
 
-    mLastOOCPos = me->GetPosition();
-    me->StopMoving();//force stop
+    me->StopMoving();
+    me->GetMotionMaster()->MovementExpired(false);
```

Comment:
Check console for "trying to delete its current MG" errors (one of the ways to reset MotionMaster caused it, not sure which one)